### PR TITLE
Implement interrupt

### PR DESCRIPTION
**Pull request by schwitzguebel**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/utils/GenericThreadLauncher.java [hunk 1]
```diff
@@ -90,7 +90,8 @@ private boolean checkInterruption(Exception exception) {
         boolean isInterrupt = false;
```
Comment (TheoPascoli) [inline gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/utils/GenericThreadLauncher.java:90]: Can you rename it "isInterrupted" please ?

**Review by TheoPascoli — CHANGES_REQUESTED** [review]

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/utils/GenericThreadLauncher.java (hunk 1 above)
Comment (schwitzguebel) [reply gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/utils/GenericThreadLauncher.java:90]: ok will do

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java [hunk 2]
```diff
@@ -47,27 +48,31 @@ public SweResponse launchDichotomy(SweData sweData) {
...
+        return new ExecutionResult<>(results);
     }
 
-    private SweDichotomyResult waitAndGet(CompletableFuture<SweDichotomyResult> dichotomy, DichotomyDirection direction) {
+    private SweDichotomyResult waitAndGet(Future<SweDichotomyResult> dichotomy) throws InterruptedException {
         try {
             return dichotomy.get();
         } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
+            throw e;
```
[the remark this answers is not in the document]
Comment (schwitzguebel) [reply gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java:72]: quite right

File: gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java [hunk 3]
```diff
@@ -47,27 +48,31 @@ public SweResponse launchDichotomy(SweData sweData) {
...
+        try {
+            futures.add(worker.runDichotomyForOneDirection(sweData, DichotomyDirection.ES_FR));
+            futures.add(worker.runDichotomyForOneDirection(sweData, DichotomyDirection.FR_ES));
+            futures.add(worker.runDichotomyForOneDirection(sweData, DichotomyDirection.ES_PT));
+            futures.add(worker.runDichotomyForOneDirection(sweData, DichotomyDirection.PT_ES));
+            for (Future<SweDichotomyResult> future : futures) {
+                results.add(waitAndGet(future));
+            }
+        } catch (InterruptedException e) {
+            futures.stream().forEach(f -> f.cancel(true));
```
[the remark this answers is not in the document]
Comment (schwitzguebel) [reply gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java:62]: will do

**Review by TheoPascoli — APPROVED** [review]

**Pull request merged**